spark-commons/src/main/scala/za/co/absa/spark/commons/errorhandling/implementations/ErrorMessageArray.scala

```diff
@@ -0,0 +1,73 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.spark.commons.errorhandling.implementations
+
+import org.apache.spark.sql.{Column, DataFrame}
+import org.apache.spark.sql.functions.{array, array_except, array_union, col, map_from_arrays, map_keys, map_values, struct, when}
+import za.co.absa.spark.commons.adapters.TransformAdapter
+import za.co.absa.spark.commons.errorhandling.partials.EvaluateIntoErrorMessage.FieldNames._
+import za.co.absa.spark.commons.errorhandling.partials.{ErrorHandlingCommon, EvaluateIntoErrorMessage}
+import za.co.absa.spark.commons.sql.functions.null_col
+import za.co.absa.spark.commons.implicits.DataFrameImplicits.DataFrameEnhancements
+
+case class ErrorMessageArray(errorColumnName: String = ErrorMessageArray.defaultErrorColumnName)
+  extends ErrorHandlingCommon
+  with EvaluateIntoErrorMessage
+  with TransformAdapter {
+
+  private def decomposeMap(errorMessageColumn: Column): Column = {
+    when(errorMessageColumn.isNotNull,
+      struct(
+        errorMessageColumn.getField(errType) as errType,
+        errorMessageColumn.getField(errCode) as errCode,
+        errorMessageColumn.getField(errMsg) as errMsg,
+        map_keys(errorMessageColumn.getField(errColsAndValues)) as errCols,
+        map_values(errorMessageColumn.getField(errColsAndValues)) as errValues,
+        errorMessageColumn.getField(additionInfo) as additionInfo
+      )
+    )
+  }
+
+  private def recomposeMap(errorMessageColumn: Column): Column = {
+    struct(
+      errorMessageColumn.getField(errType) as errType,
+      errorMessageColumn.getField(errCode) as errCode,
+      errorMessageColumn.getField(errMsg) as errMsg,
+      map_from_arrays(errorMessageColumn.getField(errCols), errorMessageColumn.getField(errValues)) as errColsAndValues ,
+      errorMessageColumn.getField(additionInfo) as additionInfo
+    )
+  }
+
+  private def deMap(arrayCol: Column): Column = transform(arrayCol, decomposeMap)
+  private def reMap(arrayCol: Column): Column = transform(arrayCol, recomposeMap)
+
+  private def appendToErrCol(dataFrame: DataFrame, errorColName: String, colToUnion: Column): DataFrame = {
+    dataFrame.withColumn(errorColName, reMap(array_union(deMap(col(errorColName)), colToUnion)))
+  }
+
+  protected def doTheColumnsAggregation(dataFrame: DataFrame, errCols: Column*): DataFrame = {
+    val aggregated = array(errCols.map(decomposeMap): _*) //need to decompose the map field, as it's not supported in array functions
+    val aggregatedWithoutNulls = array_except(aggregated, array(null_col))
+    val joinToExisting: (DataFrame, String) => DataFrame = appendToErrCol(_, _, aggregatedWithoutNulls)
+    dataFrame.withColumnIfDoesNotExist(joinToExisting)(errorColumnName, reMap(aggregatedWithoutNulls))
+  }
+
+}
+
+object ErrorMessageArray {
+  final val defaultErrorColumnName = "errCol"
```

Comment:
isn't this pretty 'standard' or 'common' throughout the errorhandling module? I know that  this might be changed for our users and maybe even multiple error columns can be supported, depending on a concrete implementation, but I think that we could find a better place for this. What do you think?

Reply:
It's only common for implementations that will add a new custom column to the `DataFrame`. So far this is the only implementation that does that. 🤷‍♂️

Reply:
I understand, but this implementation is very specific and this name is quite general for other places. But okay we can refactor it later when we have another place that would be using it as well